gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -676,6 +690,10 @@ protected void forWorkUnit(WorkUnit workUnit) {
       workUnit.setId(taskId);
       workUnit.setProp(ConfigurationKeys.TASK_ID_KEY, taskId);
       workUnit.setProp(ConfigurationKeys.TASK_KEY_KEY, Long.toString(Id.Task.parse(taskId).getSequence()));
+      // In case of MultiWorkUnit, WorkUnit(s) should be assigned different ids
+      if (workUnit instanceof MultiWorkUnit) {
+        ((MultiWorkUnit) workUnit).getWorkUnitsStream().transform(this);
```

Comment:
Is it valid to have a `MultiWorkUnit` inside another `MultiWorkUnit`? I believe this case is already taken care of by `MultiWorkUnitForEach`.